### PR TITLE
Y2K-1472 RichButton Component

### PR DESCRIPTION
**Pull request by Daniel-Wang**
Screen Shot 2020-02-20 at 2 26 11 PM

cc @keybase/react-hackers @keybase/y2ksquad

### CODE REVIEW

File: shared/common-adapters/rich-button.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+const Kb = {
+  RichButton,
+}
+
+const load = () => {
+  Sb.storiesOf('Common/RichButton', module).add('Basic', () => (
+    <Kb.RichButton
+      onClick={() => {
+        console.log('hi')
```
Comment (buoyad) [inline shared/common-adapters/rich-button.stories.tsx]: You can do `onClick={Sb.action('onClick')}` and it'll show up in the actions panel on desktop

**Comment by chrisnojima** [comment]
is this actually a common thing or just used in one place

File: shared/common-adapters/rich-button.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import * as Styles from '../styles'
+
```
Comment (buoyad) [inline shared/common-adapters/rich-button.tsx]: :fire:

File: shared/common-adapters/rich-button.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+const Kb = {
+  Avatar,
+  Box2,
+  ClickableBox,
+  Text,
+}
+
+const RichButton = (props: Props) => {
+  const [isPressing, setPressing] = React.useState()
```
Comment (buoyad) [inline shared/common-adapters/rich-button.tsx]: initialize this to something

File: shared/common-adapters/rich-button.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
+import * as Styles from '../styles'
+
+import {Box2} from './box'
+import ClickableBox from './clickable-box'
+import Avatar from './avatar'
+import Text from './text'
+
+type Props = {
+  imageUrl: string
```
Comment (buoyad) [inline shared/common-adapters/rich-button.tsx]: This can be `IconType` from `./icon`, our illustrations become icon types

**Comment by Daniel-Wang** [comment]
Used a lot in wizard, there was a note to explicitly make it a component

File: shared/common-adapters/rich-button.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  return (
+    <Kb.ClickableBox
+      className="hover_container"
+      style={Styles.collapseStyles([styles.containerStyle, isPressing && styles.mobileContainer])}
+      onClick={props.onClick}
+      onPressIn={() => setPressing(true)}
+      onPressOut={() => setPressing(false)}
+      hoverColor={Styles.globalColors.blueLighter_20}
+    >
+      <Kb.Avatar imageOverrideUrl={props.imageUrl} style={styles.thumbnail} size={64} />
```
Comment (buoyad) [inline shared/common-adapters/rich-button.tsx]: `Kb.Icon`

File: shared/common-adapters/rich-button.tsx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    backgroundColor: Styles.globalColors.blueLighter_20,
+  },
+  mobileTitle: {
+    color: Styles.globalColors.blueDark,
+  },
+  thumbnail: {
+    marginRight: Styles.globalMargins.small,
+  },
+  title: {
+    marginBottom: Styles.globalMargins.xtiny,
```
Comment (buoyad) [inline shared/common-adapters/rich-button.tsx]: Use `Box2`'s `gap` prop instead of this

**Pull request merged**